ci/azure-pipelines-dev.yml

```diff
@@ -109,6 +109,34 @@ stages:
               invoke lint || EXIT_STATUS=$?
               exit $EXIT_STATUS
 
+  - stage: unit_tests
+    dependsOn: scope_check
+    pool:
+      vmImage: 'ubuntu-latest'
+
+    jobs:
+      - job: run_unit_tests
+        condition: eq(stageDependencies.scope_check.changes.outputs['CheckChanges.GXChanged'], true)
+        steps:
+          - task: UsePythonVersion@0
+            inputs:
+              versionSpec: '3.7'
+            displayName: 'Use Python 3.7'
+
+          - bash: python -m pip install --upgrade pip==21.3.1
+            displayName: 'Update pip'
+
+          - script: |
+              pip install --constraint constraints-dev.txt ".[test]" pytest-azurepipelines
+            displayName: 'Install dependencies'
+
+          - script: |
+              invoke tests --cloud --unit --timeout=3.0
```

Comment:
I removed the `--ci` flag. We don't actually have code coverage set up in this pipeline so I dropped this flag and didn't add the publish jobs since the don't go anywhere. I also seem to recall that if one publishing coverage multiple times isn't supported.